### PR TITLE
Add dynamic http endpoints

### PR DESCRIPTION
**Pull request by iilyak**

### CODE REVIEW

File: test/chttpd_endpoints_tests.erl [hunk 1]
```diff
@@ -0,0 +1,172 @@
+-module(chttpd_endpoints_tests).
```
Comment (kxepal) [inline test/chttpd_endpoints_tests.erl]: License header.

**Comment by rnewson** [comment]
I really like the shape of this. +1 to merge after squashing the license header in.

**Comment by kxepal** [comment]
Erm..just noticed: why chttpd tests are defined here?

**Comment by iilyak** [comment]
@kxepal: I thought they are integration tests but maybe they aren't. Let me know if you want me to move them into `chttpd`.

**Comment by kxepal** [comment]
@iilyak mmm..looks like something in between. Ok, let's put them here. Need to handle this question later.

**Comment by iilyak** [comment]
They test various endpoints including the ones defined outside chttpd application.

**Comment by iilyak** [comment]
PR is squashed

**Comment by iilyak** [comment]
I'll move `_oauth` into here and squash again

**Pull request merged**